node/tessel.js

```diff
@@ -287,9 +287,21 @@ Pin.prototype.output = function output(initialValue, cb) {
     return this;
 }
 
-function I2C(addr, port) {
-    this.addr = addr;
+function I2C(params, port) {
+    this.addr = params.addr;
     this._port = port;
+    this._freq = params.freq ? params.freq : 100000; // 100khz
+    this._mode = params.mode ? params.mode : I2CSettings.Master;
+
+    // 15ns is max scl rise time
+    // f = (48e6)/(2*(5+baud)+48e6*1.5e-8)
+    this._baud = Math.floor(((48e6/this._freq) - 48e6*(1.5e-8))/2 - 5);
+    if (this._baud > 255 || this._baud <= 0) {
+        throw new Error('I2C frequency should be between 4Mhz and 90khz');
```

Comment:
We may have to do something special to support I2C fast mode (speed > 400kHz), so let's disallow that for now.